### PR TITLE
[toolchain] nightly-2022-01-01

### PR DESCRIPTION
**Pull request by PsiACE**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Happy New Year! 

- toolchain: nightly-2022-01-01

We need to have a routine maintenance event that includes at least the following.

- update the toolchain
- update dependencies that have been missed
- check that our utilities are working properly

I think once in each natural month is a relatively appropriate frequency. This allows us to check between release channels and get something fresh.

## Changelog

- Not for changelog (changelog entry is not required)

## Related Issues

Fixes #issue

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: .github/actions/build_debug/action.yml [hunk 1]
```diff
@@ -18,6 +18,8 @@ runs:
     - uses: actions-rs/toolchain@v1
       with:
         target: ${{ matrix.config.target }}
+        toolchain: nightly
```
Comment (BohuTANG) [inline .github/actions/build_debug/action.yml:21]: We have defined the nightly in the rust-toolchain, why here still need? :thinking:

File: common/clickhouse-srv/src/types/block/mod.rs [hunk 2]
```diff
@@ -240,12 +240,14 @@ impl<K: ColumnType> Block<K> {
     }
 
     /// Add new column into this block
+    #[must_use]
```
Comment (BohuTANG) [inline common/clickhouse-srv/src/types/block/mod.rs:243]: What's the `[must_use]` attribute for

File: .github/actions/build_debug/action.yml (hunk 1 above)
Comment (PsiACE) [reply .github/actions/build_debug/action.yml:21]: emm, rust-toolchain.toml hints at the format of the file, which I think is better. Unfortunately, action-rs/toolchain does not seem to support direct processing and will prompt that the toolchain file cannot be found.

File: common/clickhouse-srv/src/types/block/mod.rs (hunk 2 above)
Comment (PsiACE) [reply common/clickhouse-srv/src/types/block/mod.rs:243]: https://rust-lang.github.io/rust-clippy/master/index.html#return_self_not_must_use

> Methods returning Self often create new values, having the #[must_use] attribute prevents users from “forgetting” to use the newly created value.

**Comment by PsiACE** [comment]
emm, `servers::clickhouse::test_clickhouse_insert_to_fuse_table` is a flaky test, and I will be re-running the ci

**Comment by BohuTANG** [comment]
The CI failed is not related to this change, let's merge.

**Pull request merged**